blockprod/src/detail/mod.rs

```diff
@@ -180,15 +278,17 @@ impl BlockProduction {
     /// remnants in the job manager.
     pub async fn produce_block(
         &self,
+        input_data: Option<GenerateBlockInputData>,
         reward_destination: Destination,
         transactions_source: TransactionsSource,
     ) -> Result<(Block, oneshot::Receiver<usize>), BlockProductionError> {
-        self.produce_block_with_custom_id(reward_destination, transactions_source, None)
+        self.produce_block_with_custom_id(input_data, reward_destination, transactions_source, None)
             .await
     }
 
     async fn produce_block_with_custom_id(
         &self,
+        input_data: Option<GenerateBlockInputData>,
         _reward_destination: Destination,
```

Comment:
should `reward_destination` be used in PoW?

Reply:
Yep, this is what I wasn't too sure of. Currently there's a `Destination` parameter in the RPC call, but was thinking this may not be needed for PoS now, so it could be pushed into the `PoW` input data... I'll work out the mechanics as I make tests